### PR TITLE
Model improvements

### PR DESCRIPTION
**Pull request by gabrieltseng**
- [x] Pass the previous year's VCI (for the same target month) as static input to the model
- [x] Allow the model to predict normalized values, so it doesn't need to learn mean and std

### CODE REVIEW

File: src/models/data.py [hunk 1]
```diff
@@ -42,6 +43,16 @@ def concatenate(self, x: TrainData):
                     newval = torch.cat((val, getattr(x, key)), dim=0)
                 setattr(self, key, newval)
 
+    def filter(self, filter_array: Union[np.ndarray, torch.Tensor]) -> None:
+        # noqa because we just want to make sure these to have the same type,
```
Comment (tommylees112) [inline src/models/data.py]: what's noqa?

File: src/models/data.py [hunk 2]
```diff
@@ -42,6 +43,16 @@ def concatenate(self, x: TrainData):
                     newval = torch.cat((val, getattr(x, key)), dim=0)
                 setattr(self, key, newval)
 
+    def filter(self, filter_array: Union[np.ndarray, torch.Tensor]) -> None:
```
Comment (tommylees112) [inline src/models/data.py]: What is this function doing?

File: src/models/data.py [hunk 3]
```diff
@@ -420,6 +443,33 @@ def __init__(self, loader: DataLoader) -> None:
     def __iter__(self):
         return self
 
+    def _get_prev_y_var(
```
Comment (tommylees112) [inline src/models/data.py]: is there any way we can pass in the mean of all the previous months? So in the train data this will be either 1 (for 1982) or 29 for 2010 (29 previous years of that month).

In the test it would just be the mean of the training data (like a climatology)

Comment (tommylees112) [reply src/models/data.py]: maybe for a future PR?

Comment (tommylees112) [reply src/models/data.py]: maybe that should be `get_mean_y_value` or something

File: src/models/data.py (hunk 1 above)
Comment (gabrieltseng) [reply src/models/data.py]: Tells flake8 to ignore the line

File: src/models/data.py (hunk 2 above)
Comment (gabrieltseng) [reply src/models/data.py]: Before, we had a bunch of lines which were
`historical = historical[notnan_indices]`

now, we can filter the entire `TrainData` object, so

`traindata.filter(notnan_indices)`

File: src/models/data.py (hunk 3 above)
Comment (gabrieltseng) [reply src/models/data.py]: yea. We sort of have that with the yearly_aggs, but it could definitely be revisited.

File: src/models/data.py (hunk 2 above)
Comment (tommylees112) [reply src/models/data.py]: gotcha
